### ISSUE TITLE
Repositories added in bulk are not automatically detected as GitHub/non-GitHub

### ISSUE THREAD

**Issue by Kwpolska** [desktop/desktop#1680]
### Description

After dragging and dropping all my git repositories to GitHub Desktop, they were all displayed as “Other” repos until I opened all of them manually — only then were they separeted into GitHub, and Other repos.

### Version

**GitHub Desktop version:** 0.5.5
**OS version:** macOS Sierra 10.12.5

### Steps to Reproduce

1. Clone all repositories you work with to some directory.
2. Drag-and-drop all files in that directory to GitHub Desktop for import.
3. Observe that all of them appear under “Other” (:desktop_computer:) and become “GitHub” (:blue_book:) only when opened manually (unless #1679.)

**Reproduces how often:** always

### Logs

```
2017-05-21T08:39:59.138Z - error: fetchRepository: not found for 'Kwpolska' and 'Kwpolska/staystatic'
Error: {
  "message": "Not Found",
  "documentation_url": "https://developer.github.com/v3"
}

    at file:///Applications/GitHub%20Desktop.app/Contents/Resources/app/renderer.js:3:78193
    at XMLHttpRequest.J.ae.onreadystatechange (file:///Applications/GitHub%20Desktop.app/Contents/Resources/app/renderer.js:3:75070)
2017-05-21T08:41:02.072Z - error: fetchRepository: not found for 'Kwpolska' and 'Kwpolska/python-project-template.wiki'
Error: {
  "message": "Not Found",
  "documentation_url": "https://developer.github.com/v3"
}

    at file:///Applications/GitHub%20Desktop.app/Contents/Resources/app/renderer.js:3:78193
    at XMLHttpRequest.J.ae.onreadystatechange (file:///Applications/GitHub%20Desktop.app/Contents/Resources/app/renderer.js:3:75070)
```

**Comment by shiftkey** [comment]
This is related to [`refreshGitHubRepositoryInfo `](https://github.com/desktop/desktop/blob/89b71a704007cd62b45db6ebdb279abd8deadc0b/app/src/lib/dispatcher/dispatcher.ts#L179).

> Observe that all of them appear under “Other” (🖥) and become “GitHub” (📘) only when opened manually (unless #1679.)

I thought [this usage](https://github.com/desktop/desktop/blob/89b71a704007cd62b45db6ebdb279abd8deadc0b/app/src/lib/dispatcher/dispatcher.ts#L157) would fire when you drag-and-drop the repositories onto the app, but it sounds like [this usage](https://github.com/desktop/desktop/blob/89b71a704007cd62b45db6ebdb279abd8deadc0b/app/src/lib/dispatcher/dispatcher.ts#L241) is what you're seeing. 
 
> error: fetchRepository: not found for 'Kwpolska' and 'Kwpolska/staystatic'
> error: fetchRepository: not found for 'Kwpolska' and 'Kwpolska/python-project-template.wiki'

These errors are from the GitHub API saying "I can't find a repository matching this name under your  account", which seems like the correct behaviour. Without API information, we can't group this under the GitHub tab, and there's probably other issues if the remote no longer exists for these repositories.

**Comment by Kwpolska** [comment]
> I thought this usage would fire when you drag-and-drop the repositories onto the app, but it sounds like this usage is what you're seeing.

(I have no idea what that means and what the difference is.)

I suspect this: https://github.com/desktop/desktop/blob/master/app/src/lib/remote-parsing.ts#L26 — it assumes repository links end with `.git`, which does not have to be true.

> These errors are from the GitHub API saying "I can't find a repository matching this name under your account", which seems like the correct behaviour. Without API information, we can't group this under the GitHub tab, and there's probably other issues if the remote no longer exists for these repositories.

`staystatic` was deleted, but `python-project-template.wiki` wasn’t: https://github.com/Kwpolska/python-project-template/wiki

Perhaps that’s another bug regarding detection of GitHub Wikis.

**Comment by shiftkey** [comment]
> (I have no idea what that means and what the difference is.)

Lazy checks versus eager checks. Just a note for whoever investigates this to look at those two usages.

> Perhaps that’s another bug regarding detection of GitHub Wikis.

Probably, we've got #1624 open to look at the cloning side.

**Comment by Janpot** [comment]
Maybe related? Every time I add a local repository I have to change the remote from `https://janpot@github.com/janpot/some-repo.git` to `https://github.com/janpot/some-repo.git`
Otherwise it doesn't show up as a "github" repo, neither does the authentication work. I think sourcetree used to set the remote up like that.

**Comment by joshaber** [comment]
It looks like this is essentially a duplicate of #1679 so I'm gonna close it. Let me know if I've misunderstood!

**Comment by Kwpolska** [comment]
Please reopen this, and compare the authors of the two issues.

In this issue, GitHub does not automatically detect the source of a repository after importing them in bulk, and I need to manually open each repo to hopefully have them detected. However, this sometimes fails with remotes that don’t end in git, and that’s what #1679 is about.

**Comment by joshaber** [comment]
Ah, I see, thanks for the clarification!

**Comment by iAmWillShepherd** [comment]
@Kwpolska I'm unable to reproduce this. Are you still experiencing this on the latest version?

**Comment by Kwpolska** [comment]
Yes. Here’s a screencast: https://www.dropbox.com/s/lddg8iuj46nzod1/github-desktop-1680.mov?dl=0

**Comment by iAmWillShepherd** [comment]
@Kwpolska , I've been trying to reproduce this for a couple of hours and have had no luck. 

This is the behavior I get when running through your steps.
![2017-06-09 13_50_59](https://user-images.githubusercontent.com/1715082/26990005-d00954f4-4d1a-11e7-9fce-f5e2f9486ef9.gif)

Is there anything unique about the repos you are attempting to add?

**Comment by Kwpolska** [comment]
Not really. I’ve had some issues with SSH urls in this app, so maybe that’s the problem?

**Comment by joshaber** [comment]
We fixed some issues around SSH URLs in 0.6.0. Could you try updating to see if you can still reproduce it with those fixes?

**Comment by Kwpolska** [comment]
It works!
